articles/frameworks/vue.md

```diff
@@ -1,3 +1,104 @@
 # Vue.js support
 
-TODO
+Visual Studio improved support for [Vue.js](https://vuejs.org/) framework, which enhances the experience when developing an application with Vue.js, JavaScript and TypeScript.
+
+## Feature overview
+
+In addition to the normal Visual Studio features when writing TypeScript or JavaScript, you'll see new behavior for Vue.js specific features:
+
+* Supported Script, Style and Template blocks in .vue files.
+* Recognize lang attribute on .vue files.
+
+## Create a Vue.js project with the vue-cli
+
+Vue.js provides and official CLI for quickly scaffolding projects. If you'll like to use it to create your application, follow the steps below to setup your developer environment.
+
+> [!NOTE]
+> This steps assumes you already have some knowledge about the Vue.js framework. If not, we suggest to visit [Vue.js](https://vuejs.org/) for details about it.
+
+### Create a new Visual Studio project
+
+We use an Empty ASP.NET Core Application (C#) for this example, but you can choose from a variety of projects and languages of your choice.
+
+##### Create an Empty project:
+1. Open Visual Studio and choose **File > New > Project** from the main menu.
+1. Under **Visual C# > Web**, choose **ASP.NET Core Web Application** and click OK.
+1. Select **Empty**, click OK.
+
+##### Configure the project startup:
+1. Open the file **./Startup.cs**, and add the following lines to the Configure method:
+
+```c#
+app.UseDefaultFiles() // Enables default file mapping on the web root.
+app.UseStaticFiles(); // Marks files on the web root as servable.
+```
+
+### Install the vue-cli
+
+To install the vue-cli, open a Command Prompt and type `npm install --g vue-cli` or `npm install -g @vue/cli` for version 3.0 (currently in beta).
+
+#### Use the vue-cli to scaffold a new client application:
+1. Go to your Command Prompt and change directory to your project root folder.
+1. Type `vue init webpack ClientApp` and follow the additional questions.
+
+##### Tweak the webpack config to output the built files to wwwroot:
+1. Open the file **./ClientApp/config/index.js**, and change the `build.index` and `build.assetsRoot` to wwwroot path:
+
+```js
+// Template for index.html
+index: path.resolve(__dirname, '../../wwwroot/index.html'),
+
+// Paths
+assetsRoot: path.resolve(__dirname, '../../wwwroot'),
+```
+
+##### Indicate the project to build the ClientApp everytime building is triggered:
+1. Go to **Project properties > Build Events**
+1. On **Pre-build event command line** type `npm --prefix ./ClientApp run build`.
```

Comment:
What does the `--prefix` option do?

Reply:
Ah, OK. https://stackoverflow.com/a/14469516/1674945

Reply:
(CC @uniqueiniquity who was asking about this earlier :smile:)

Reply:
@billti I was trying that, but I don't think it behaves exactly as you would expect it to when used with `npm install`. That's a discussion for a different time and place though :)